components/brave_wallet_ui/common/async/lib.ts

```diff
@@ -182,6 +179,10 @@ export function refreshBalances (currentNetwork: BraveWallet.EthereumChain) {
       balances: getBalanceReturnInfos
     }))
 
+    const visibleAssets: BraveWallet.BlockchainToken[] = visibleTokensInfo.tokens.length === 0 ? [nativeAsset] : visibleTokensInfo.tokens
+    await dispatch(WalletActions.setVisibleTokensInfo(visibleAssets))
+    const visibleTokens = visibleAssets.filter(asset => asset.contractAddress !== '')
```

Comment:
The native asset needs to be re-added after the RPC request in this next method here https://github.com/brave/brave-core/pull/12046/files#diff-b85091d7cf18c04a6a4ddd6a277db8bbe711a12299e16e98c0a3e7e11f96c00eR186-R193

Since it then is passed to `tokenBalancesUpdated` which uses an `index` i believe to set values in the reducer and will cause an issue like so.

https://user-images.githubusercontent.com/40611140/151629970-6c7e9e86-4b63-41eb-b7ce-504e2ccad02e.mov

Reply:
Yes, the token indexes were off by 1. Fixed it by applying the same filter on visible assets in the reducer.